src/Arcus.Security.Providers.AzureKeyVault/Authentication/ServicePrincipalAuthenticator.cs

```diff
@@ -7,13 +7,16 @@
 
 namespace Arcus.Security.Providers.AzureKeyVault.Authentication
 {
-    public class ServicePrincipalAuthenticator: IKeyVaultAuthenticator
+    /// <summary>
+    /// Representation of an <see cref="IKeyVaultAuthentication"/> that will generate a <see cref="IKeyVaultClient"/> implementation using a service principle.
+    /// </summary>
+    public class ServicePrincipalAuthenticator : IKeyVaultAuthentication
```

Comment:
You'll need to implement both actually, otherwise, this is a breaking change

Reply:
Ah, yes; hmm; then I have to rename `Authenticate`.

Reply:
Not really